app/src/main/java/fly/speedmeter/grub/MainActivity.java

```diff
@@ -326,6 +357,25 @@ public void onBackPressed(){
         startActivity(a);
     }
 
+    public void updateIsMPH(){
+        if(isMPH == sharedPreferences.getBoolean("isMPH", false))
+            return;
+
+        isMPH = sharedPreferences.getBoolean("isMPH", false);
+
+        if(isMPH)
+            data.setMaxSpeed(data.getMaxSpeedDouble() * 0.621371);
+        else
+            data.setMaxSpeed(data.getMaxSpeedDouble() * 1.60934);
```

Comment:
I don't understand what is happening here -- SpeedMeter should always store speeds in kilometers per second and only display in miles per hour.